src/app/views/auth.go

```diff
@@ -417,4 +417,84 @@ func authGroup(router *gin.Engine) {
 
 	})
 
+	g.POST("/socious", auth.SSOLoginRequired(), func(c *gin.Context) {
+		u, _ := c.Get("user")
+		ctx, _ := c.Get("ctx")
+		tokenClaims, _ := c.Get("token_claims")
+
+		//Castings
+		ssoClaims := tokenClaims.(auth.SSOClaims)
+		var user *models.User
+
+		//If user doesn't exist, it needs to be registered
+		if u == nil {
+			//
+			newUser := &models.User{
+				Email:     *ssoClaims.Email,
+				FirstName: ssoClaims.FirstName,
+				LastName:  ssoClaims.LastName,
+				Username:  auth.GenerateUsername(*ssoClaims.Email),
+			}
+
+			if err := newUser.Create(ctx.(context.Context)); err != nil {
+				c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+				return
+			}
+			newUser.Status = "ACTIVE"
+			if err := newUser.Verify(ctx.(context.Context)); err != nil {
+				c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+				return
+			}
+			user = newUser
+		} else {
+			user = u.(*models.User)
+		}
+
+		tokens, err := auth.GenerateFullTokens(user.ID.String())
+
+		if user.Password != nil {
+			tokens["status"] = "COMPLETED"
+		} else {
+			tokens["status"] = "PASSWORD_NOT_SET"
+		}
+
+		if err != nil {
+			c.JSON(http.StatusBadRequest, tokens)
+		}
+		c.JSON(http.StatusOK, tokens)
+
+	})
+
+	g.POST("/socious/set-password", auth.LoginRequired(), func(c *gin.Context) {
```

Comment:
what is this ?

Reply:
After the user has been registered, we may or may not ( it's your decision) prompt him/her to set a password for later simple login with email and password ( because the password field would be null and i thought it might break the flow )
should i remove it?